src/olympia/abuse/models.py

```diff
@@ -475,10 +476,18 @@ def report(self):
         self.update(job_id=job_id)
 
     def process_decision(self, *, decision_id, decision_date, decision_actions):
+        if (
+            not decision_actions
+            or len(decision_actions) > 1
+            or not self.DECISION_ACTIONS.has_api_value(decision_actions[0])
```

Comment:
IMHO this method shouldn't be dealing with api values, that should be done by the view, much like we process the timestamp before passing `decision_date`. Let the view worry about API stuff, and when it comes to doing the processing we should be dealing with values that are already valid choices for each field only.

Reply:
https://github.com/mozilla/addons-server/pull/21411/commits/0c2b29e5ad53546681d6ef6f9f3004f9066a1f00
(and refactored the view to be flatter with exceptions raised to escape because the level of indentation was getting a little OTT)